### PR TITLE
Provide testing implementation of time.Now()

### PR DESCRIPTION
**Pull request by lbernick**
# Changes
Prior to this commit, the TaskRun and PipelineRun reconciler tests implicitly depended on the value of time.Now.
This commit allows tests to use a specific time as Now when reconciling TaskRuns and PipelineRuns.
No functional changes.

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by lbernick** [comment]
may also address #2992

**Review by vdemeester — APPROVED** [review]

File: cmd/controller/main.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/clock"
```
Comment (mattmoor) [inline cmd/controller/main.go:27]: @vdemeester @lbernick  We used to have our own `system.Clock` in knative.dev/pkg for exactly this reason and we moved to `k8s.io/apimachinery/pkg/util/clock` instead.

Comment (mattmoor) [reply cmd/controller/main.go:27]: It looks like this could be a drop-in replacement here.

Comment (vdemeester) [reply cmd/controller/main.go:27]: 👍🏼

**Pull request merged**